### PR TITLE
Changes on Ballerina action invocation

### PR DESCRIPTION
**Pull request by Biruntha**

### CODE REVIEW

**Comment by hasithaa** [comment]
Tested this PR on top of https://github.com/ballerinalang/ballerina/pull/3071. No issue found.

File: amazonlambda/src/org/wso2/ballerina/connectors/amazonlambda/amazonlambda.bal [hunk 1]
```diff
@@ -5,146 +5,139 @@ import ballerina.doc;
...
         return response;
     }
 
-    @doc:Description{ value : "Lists all versions of a amazon lambda function"}
-    @doc:Param{ value : "amz: The amazonLambda connector instance"}
-    @doc:Return{ value : "response message"}
-    action getFunctionVersions(ClientConnector amz, string arn) (message) {
+    @doc:Description { value:"Lists all versions of a amazon lambda function"}
+    @doc:Return { value:"response message"}
+    action getFunctionVersions (string arn) (message) {
```
Comment (hasithaa) [inline amazonlambda/src/org/wso2/ballerina/connectors/amazonlambda/amazonlambda.bal]: missing doc annotation for arn

File: amazonlambda/src/org/wso2/ballerina/connectors/amazonlambda/amazonlambda.bal [hunk 2]
```diff
@@ -5,146 +5,139 @@ import ballerina.doc;
...
 
-    @doc:Description{ value : "Gets the list of functions and their configuration information"}
-    @doc:Param{ value : "amz: The amazonLambda connector instance"}
-    @doc:Param{ value : "arn: The amazon resource name of the function to invoke"}
-    @doc:Return{ value : "response message"}
-    action listFunctions(ClientConnector amz) (message) {
+    @doc:Description { value:"Gets the list of functions and their configuration information"}
+    @doc:Param { value:"arn: The amazon resource name of the function to invoke"}
+    @doc:Return { value:"response message"}
+    action listFunctions () (message) {
```
Comment (hasithaa) [inline amazonlambda/src/org/wso2/ballerina/connectors/amazonlambda/amazonlambda.bal]: There is no parameter called arn in the action.

**Comment by hasithaa** [comment]
This PR is depends on https://github.com/ballerinalang/ballerina/pull/3100

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 3]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "GET";
+        requestURI = "/";
+        host = "s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 4]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "GET";
+        requestURI = "/";
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 5]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "PUT";
+        requestURI = "/";
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 6]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "DELETE";
+        requestURI = "/";
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 7]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "GET";
+        requestURI = "/" + objectName;
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 8]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "DELETE";
+        requestURI = "/" + objectName;
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

File: amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal [hunk 9]
```diff
@@ -4,212 +4,205 @@ import org.wso2.ballerina.connectors.amazonauth;
...
+        string requestURI;
+        string host;
+        string endpoint;
+        message requestMsg = {};
+        message response;
+
+        httpMethod = "PUT";
+        requestURI = "/" + objectName;
+        host = bucketName + ".s3.amazonaws.com:443";
+        endpoint = "https://" + host;
```
Comment (hasithaa) [inline amazons3/src/org/wso2/ballerina/connectors/amazons3/amazonS3Connector.bal]: Please assign above variables in-line.

**Review by hasithaa — APPROVED** [review]

**Review by sameerajayasoma — APPROVED** [review]

**Pull request merged**